library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
Please, read it again. There is no linear search over the whole index. At all. Linear search is performed over 2-3 entries at most.

I'm not going to argue more trying to prove this, didn't I show you the code and write explanation?
If you do not want to understand the code you are trying to change, I can not accept such pull requests.
If you have any further questions on how eblob was designed you are welcome, but I will not repeat myself again.